sqlglot/dialects/doris.py

```diff
@@ -11,6 +11,12 @@
 from sqlglot.dialects.mysql import MySQL
 
 
+def _tonumber_sql(self, expression: exp.ToNumber) -> str:
+    if expression.args.get("format"):
+        self.unsupported("Format argument unsupported for TO_NUMBER function")
+    return self.sql(exp.cast(expression.this, "double"))
```

Comment:
Let's move this function to the base Generator so that other dialects can also leverage this logic.

We should have a class-level constant in the generator called `SUPPORTS_TO_NUMBER`, which defaults to `True` and is set to `False` for the dialects that don't support it - see my previous comment about which dialects support it. If it's set, we need to use `fallback_function_sql` so that `TO_NUMBER` will be correctly generated, otherwise we should use this logic.

As a final note, let's change the warning to "Unsupported TO_NUMBER function, converting to a CAST as DOUBLE".

Reply:
I have modified it and add test for all dialects, please review